test/api/notifications_mailer_test.rb

```diff
@@ -0,0 +1,86 @@
+require 'test_helper'
+
+class NotificationsMailerTest < ActionMailer::TestCase
+
+  #
+  # Init params for send mail function
+  #
+  setup do
+    @unit_role = UnitRole.first
+    @unit = @unit_role.unit
+    @convenor = @unit.main_convenor
+    @summary_stats = {
+      unit: @unit,
+      staff: {},
+      revert: {},
+      convenor: @convenor,
+      week_start: '20190908'.to_date,
+      week_end: '20190908'.to_date - 7.days,
+      num_students_without_tutors: 8,
+      revert_count: 2,
+      unit_week_comments: 1
+    }
+    @summary_stats[:staff][@unit_role] = {
+      received_comments: 10,
+      sent_comments: 5,
+      total_comments: 15,
+      tasks_awaiting_feedback_count: 4,
+      oldest_task_days: 2,
+      number_of_students: 3,
+      staff: @unit_role.user,
+      total_staff_engagements: 3,
+      staff_engagements: 6,
+    }
+    @summary_stats[:revert][@unit_role.user] = []
+  end
+
```

Comment:
How did you make sure all of the above stats are correct? Where did you create the unit with these dates, numbers of students, staff, students at that status, etc?

Reply:
I'm not sure if it is necessary because we are implementing unit test for function send email. So I thought no need take care more out of scope.